File: benchmarks/reference/java/BranchStress.java
/** BranchStress.java — Branch-heavy loop benchmark (20M iterations).
    Equivalent to examples/il/benchmarks/branch_stress.il */
public class BranchStress {
    public static void main(String[] args) {
        long n = 20000000 + args.length;
        long count = 0;
        for (long i = 0; i < n; i++) {
            if (i % 2 == 0) count += 1;
            if (i % 3 == 0) count += 2;
            if (i % 5 == 0) count += 3;
            if (i % 7 == 0) count += 5;
        }
        System.exit((int)(count & 0xFF));
    }
}

File: benchmarks/reference/lua/branch_stress.lua
-- branch_stress.lua -- Branch-heavy loop benchmark (200K iterations).
-- Equivalent to examples/il/benchmarks/branch_stress.il

local count = 0
for i = 0, 199999 do
    if i % 2 == 0 then count = count + 1 end
    if i % 3 == 0 then count = count + 2 end
    if i % 5 == 0 then count = count + 3 end
    if i % 7 == 0 then count = count + 5 end
end

os.exit(count & 0xFF)
